doc/Expressions.md

```diff
@@ -0,0 +1,385 @@
+
+## Expressions
+
+The following paragraph lists all available expressions divided into the categories: basic; mathematical; boolean logic.
+
+### Basic
+
+#### LITERAL
+
+The `LITERAL` expression has one operand that holds a static value of a [primitive type](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Primitive_values) supported by JavaScript. It is often used as fact function and can serve as operand for other expressions.
+
+> JavaScript has some natural problems with big numbers and floating points. To overcome some of this issues, numbers are handled by [big.js](https://github.com/MikeMcl/big.js/).
+
+```json
+{
+    "fact": "[aantal maanden in een jaar]",
+    "function": {
+        "expression": "LITERAL",
+        "operand": 12
+    }
+}
+```
+
+#### LIST
+
+The `LIST` operator can be used to collect data during evaluation of a fact. The fact containing this list can be used as input for other expressions, for example `SUM`.
+
+The `LIST` expression has a somewhat different syntax when compared to other expressions. It needs a unique `name` property for administrative purpose and a `items` property. The `items` property can be either a fact reference or another expressions.
+
+```json
+{
+    "fact": "[totale oppervlakte]",
+    "function": {
+        "expression": "SUM",
+        "operands": [
+            {
+                "expression": "LIST",
+                "name": "perceel",
+                "items": "[oppervlakte van het perceel]"
+            }
+        ]
+    }
+}
+```
+
+#### CREATE
+
+The `CREATE` expression can be used to make sure that a fact is created as a result of an action. This prevents the usage of facts that are defined elsewhere outside the model and gives better control over the execution flow. At the same time, it is possible to enforce that other fact must exist in the context at the time of creation. Later in the program, all these facts can be referenced with the `PROJECTION` expression.
+
+```json
+{
+    "acts": [
+        {
+            "act": "<<kinderbijslag aanvragen>>",
+            "actor": "[ouder]",
+            "preconditions": "[bedrag]",
+            "recipent": "[minister]",
+            "create": [
+                "[aanvraag]"
+            ]
+        }
+    ],
+    "facts": [
+        {
+            "fact": "[aanvraag]",
+            "function": {
+                "expression": "CREATE",
+                "operands": [
+                    "[bedrag]"
+                ]	
+            }
+        }
+    ]
+}
+```
+
+#### PROJECTION
+
+The `PROJECTION` expression can be used the retrieve the contents of facts defined with the `CREATE` expression. It accepts a `context` and `fact` property to define the fact that must be retrieved and the context in which it should be defined. If the fact could not be found, taking the action will fail.
+
+```json
+{
+    "acts": [
+        {
+            // Act that creates the '[aanvraag]' fact
```

Comment:
Are these to be filled in? If not, using jsonc as highlighting should render them nicely

Reply:
There's a typo here. used the retrieve